.github/workflows/apply.yml

```diff
@@ -0,0 +1,105 @@
+name: 'Terraform plan'
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        required: true
+        description: The environment to deploy against.
+        type: string
+        default: staging
+permissions:
+  id-token: write
+  contents: write
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      account-number-secret: ${{ steps.set-environment-names.outputs.account_number_secret }}
+      title-environment: ${{ steps.set-environment-names.outputs.title_environment }}
+    steps:
+      - id: set-environment-names
+        run: |
+          env = "${{ github.event.inputs.environment }}"
+          print(f"::set-output name=account_number_secret::{env.upper()}_ACCOUNT_NUMBER")
+          print(f"::set-output name=title_environment::{env.title()}")
+        shell: python
+  plan:
+    runs-on: ubuntu-latest
+    needs: setup
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          token: ${{ secrets.WORKFLOW_PAT }}
+      - name: Configure AWS credentials for Lambda
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.MANAGEMENT_ACCOUNT }}:role/TDRGithubTerraformAssumeRole${{ needs.setup.outputs.title-environment }}
+          aws-region: eu-west-2
+          role-session-name: TerraformRole
+      - name: Terraform Plan
+        id: plan
+        env:
+          GITHUB_OWNER: nationalarchives
+          GITHUB_TOKEN: ${{ secrets.WORKFLOW_PAT }}
+          TF_VAR_tdr_account_number: ${{ secrets[needs.setup.outputs.account-number-secret] }}
+        run: |
+            terraform init
+            terraform workspace select ${{ github.event.inputs.environment }}
+            pip install boto3
+            terraform plan -no-color -out=out > /dev/null
```

Comment:
Will the "color" plan work on GitHub. It was only set to `no-color` as the Jenkins UI couldn't cope with it

Reply:
These logs are being uploaded to Cloudwatch and I don't think Cloudwatch supports colours either so we do need this.